catalog/app/containers/Auth/SSOSignUp.js

```diff
@@ -0,0 +1,148 @@
+import { reduxForm, Field, SubmissionError } from 'redux-form/es/immutable'
+import * as reduxHook from 'redux-react-hook'
+import * as M from '@material-ui/core'
+import { unstable_Box as Box } from '@material-ui/core/Box'
+
+import { push as notify } from 'containers/Notifications/actions'
+import * as Config from 'utils/Config'
+import * as NamedRoutes from 'utils/NamedRoutes'
+import * as Sentry from 'utils/Sentry'
+import Link from 'utils/StyledLink'
+import defer from 'utils/defer'
+import parseSearch from 'utils/parseSearch'
+import * as validators from 'utils/validators'
+
+import * as Layout from './Layout'
+import * as actions from './actions'
+import * as errors from './errors'
+import msg from './messages'
+import * as selectors from './selectors'
+
+const Container = Layout.mkLayout(<FM {...msg.ssoSignUpHeading} />)
+
+const Form = reduxForm({ form: 'Auth.SSO.SignUp' })(({ children, ...props }) => (
+  <form onSubmit={props.handleSubmit}>{children(props)}</form>
+))
+
+export default ({ location: { search } }) => {
+  const { provider, token, next } = parseSearch(search)
+
+  const dispatch = reduxHook.useDispatch()
+  const { urls } = NamedRoutes.use()
+  const sentry = Sentry.use()
+  const authenticated = reduxHook.useMappedState(selectors.authenticated)
+  const cfg = Config.useConfig()
+
+  const back = React.useCallback(() => dispatch(goBack()), [dispatch])
+
+  const onSubmit = React.useCallback(
+    async (values) => {
+      const { username } = values.toJS()
+      try {
+        const result = defer()
+        dispatch(actions.signUp({ username, provider, token }, result.resolver))
+        await result.promise
+      } catch (e) {
+        if (e instanceof errors.UsernameTaken) {
+          throw new SubmissionError({ username: 'taken' })
+        }
+        if (e instanceof errors.InvalidUsername) {
+          throw new SubmissionError({ username: 'invalid' })
+        }
+        if (e instanceof errors.EmailDomainNotAllowed) {
+          throw new SubmissionError({ _error: 'emailDomain' })
+        }
+        if (e instanceof errors.SMTPError) {
+          throw new SubmissionError({ _error: 'smtp' })
+        }
+        sentry('captureException', e)
+        throw new SubmissionError({ _error: 'unexpected' })
+      }
+
+      try {
+        const result = defer()
+        dispatch(actions.signIn({ provider, token }, result.resolver))
+        await result.promise
+      } catch (e) {
+        dispatch(notify(<FM {...msg.ssoSignUpSignInError} />))
+        dispatch(push(urls.signIn(next)))
+        sentry('captureException', e)
+        throw new SubmissionError({ _error: 'unexpected' })
+      }
+    },
+    [provider, token, next, urls, dispatch],
+  )
+
+  if (authenticated) {
+    return <Redirect to={next || cfg.signInRedirect} />
+  }
+
+  return (
+    <Container>
+      <Form onSubmit={onSubmit}>
+        {({ submitting, submitFailed, invalid, error }) => (
+          <>
+            <Field
```

Comment:
Not required right now, but nice to have in the future: when `passwordAuth === true`, the user needs to enter both a username and a password in order to sign up.

(I.e., if we allow passwords at all, we shouddn't allow the user to sign up without setting a password.)

Reply:
makes sense, i'll add this

Reply:
@dimaryaz i've changed the check to be `!!passwordAuth` instead, bc it makes sense to require password even if password sign-up is disabled (but sign-in is enabled)